development/custom-image/create-custom-image.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+set -o errexit
+
+readonly CURRENT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+readonly ROOT_DIR=${CURRENT_DIR}/../../
+# shellcheck disable=SC1090
+source "${ROOT_DIR}/prow/scripts/library.sh"
+
+cleanup() {   
+    shout "Removing instance kyma-deps-image-vm-${RANDOM_ID}"
+    gcloud compute instances delete --quiet --zone "${ZONE}" "kyma-deps-image-vm-${RANDOM_ID}"
+}
+
+
+RANDOM_ID=$(openssl rand -hex 4)
+DATE=$(date +v%Y%m%d)
+ZONE=$(openssl rand -hex 4)
+
+ZONE_LIMIT=${ZONE_LIMIT:-5}
+EU_ZONES=$(gcloud compute zones list --filter="name~europe" --limit="${ZONE_LIMIT}" | tail -n +2 | awk '{print $1}')
+for ZONE in ${EU_ZONES}; do
+    shout "Attempting to create a new instance named kyma-deps-image-vm-${RANDOM_ID} in zone ${ZONE}"
+    gcloud compute instances create "kyma-deps-image-vm-${RANDOM_ID}" \
+        --metadata enable-oslogin=TRUE \
+        --machine-type n1-standard-4 \
+        --image debian-9-stretch-v20181011 \
+        --image-project debian-cloud \
+        --zone "${ZONE}" \
+        --boot-disk-size 20 &&\
+    shout "Created kyma-deps-image-vm-${RANDOM_ID} in zone ${ZONE}" && break
+    shout "Could not create machine in zone ${ZONE}"
+done
+
+trap cleanup exit
+
+shout "Moving install-deps-debian.sh to kyma-deps-image-vm-${RANDOM_ID} in zone ${ZONE} ..."
+for i in $(seq 1 5); do
+    [[ ${i} -gt 1 ]] && echo 'Retrying in 15 seconds..' && sleep 15;
+    gcloud compute scp --quiet --zone="${ZONE}" --recurse  ./install-deps-debian.sh "kyma-deps-image-vm-${RANDOM_ID}":~/ && break;
+    [[ ${i} -ge 5 ]] && echo "Failed after $i attempts." && exit 1
+done;
+
+
+shout "Running install-deps-debian.sh ..."
+gcloud compute ssh --quiet --zone="${ZONE}" "kyma-deps-image-vm-${RANDOM_ID}" -- ./install-deps-debian.sh
```

Comment:
Do we know what happens with `/etc/machine-id` during this procedure? Is it deleted? Does it persist in the template?

Reply:
I would have a request regarding this. Once you have created an image in this way please print the content of `/etc-machine-id` somewhere. Then spawn a new machine from the newly generated template. If the new machine has the same value in `/etc/machine-id` we need to clear the content of this file before we create the template